### PR TITLE
Addition of screenshot reporter utility

### PR DESCRIPTION
**Pull request by jperez1-via**
# PR Details

Screenshot reporter utility to allow generating screenshot based reporting.

## Description

Screenshot reporter utility implementation: For now, it has been created for helping on tests troubleshooting, although it could be extended and used also for formalizing execution evidences.

## Related Issue
#46

## Motivation and Context

From now on, this utility can be used to generate screenshot reporting with the purpose of helping when troubleshooting flaky test in continuous integration environment.

## How Has This Been Tested

No specific testing of this has been carried out within the library however a proof of concept has been done in my local project to ensure the Screenshot Reporter utility works as expected.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [x] My change requires a change to the documentation 
- [x] I have updated the documentation accordingly (README.md for each widget)
- [x] Increase version in package.json following [Semantic Versioning](https://semver.org/)
- [x] Add the issue into the right [project](https://github.com/systelab/systelab-components-wdio-test/projects) with the proper status (In progress)

### CODE REVIEW

File: SCREENSHOTS.md [hunk 1]
```diff
@@ -78,3 +78,44 @@ Current implementation of the utility is based on following external libraries:
 - [pngjs](https://github.com/lukeapage/pngjs): For image read/write to disk
 
 
+# Screenshot reporting
```
Comment (joaquimvila) [inline SCREENSHOTS.md:81]: Picky one: Shouldn't this title be at the same level of "generate a screenshot" or "screenshot comparison"? If that's the case, I would use 2 hashes for it

File: SCREENSHOTS.md [hunk 2]
```diff
@@ -78,3 +78,44 @@ Current implementation of the utility is based on following external libraries:
...
+### Usage: 
+
+Set the base path for the output of screenshot reporting. To be called from wdio.conf.ts beforeSession hook:
+```typescript
+beforeSession: function (config, capabilities, specs) {
+  ScreenshotReporter.setBasePath('../reports/screenshots/e2e');
+},
+```
+
+Call ScreenshotReporter.beforeSuite() method. To be called from wdio.conf.ts beforeSuite hook:
```
Comment (joaquimvila) [inline SCREENSHOTS.md:109]: I would surround method names on the text with backticks, so they are highlighed when the markdown is shown on read mode. For instance,  `ScreenshotReporter.beforeSuite()` is more readable than just ScreenshotReporter.beforeSuite(). I would apply backticks also to wdio.conf.ts file.

File: SCREENSHOTS.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # Screenshot utilities
 
-This library provides several methods to faciliate implementing screenshot-based testing techniques.
+This utility provides several methods to faciliate implementing screenshot-based testing techniques.
```
Comment (joaquimvila) [inline SCREENSHOTS.md:3]: I would say: This library provides several utilities to...

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "systelab-components-wdio-test",
-  "version": "1.9.0",
+  "version": "2.0.0",
```
Comment (joaquimvila) [inline package.json:3]: Considering this is not a breaking changes (just a new feature), new version number should be 1.10.0

**Review by joaquimvila — APPROVED** [review]
Good job @jperez1-via! This will be very helpful for troubleshooting flaky e2e tests. Just added some picky comments on the documentation and another one regarding the versioning of the package.

**Review by joaquimvila — APPROVED** [review]
Good job @jperez1-via! This will be very helpful for troubleshooting flaky e2e tests. Just added some picky comments on the documentation and another one regarding the versioning of the package.

File: src/reporters/screenshot.reporter.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    public static setBasePath(basePath: string): void {
+        this.basePath = basePath;
+    }
+
+    public static beforeSuite(suite: any): void {
+        this.specCount = 0;
+    }
+
+    public static async afterTest(test: { description: string, fullName: string }, context: any,
```
Comment (DaniMartinez-Gonzalez) [inline src/reporters/screenshot.reporter.ts:15]: Just curiosity, do you now if the context could be used to specify the browser name where the test was executed? Thanks a lot, Jordi! I think that could be very useful for the rest of the projects. :-)

**Review by DaniMartinez-Gonzalez — APPROVED** [review]
Again, thanks a lot for your work

**Review by Franek75 — APPROVED** [review]

**Pull request merged**